browser/ui/webui/settings/default_brave_shields_handler.h

```diff
@@ -35,6 +38,10 @@ class DefaultBraveShieldsHandler : public settings::SettingsPageUIHandler,
       const ContentSettingsPattern& secondary_pattern,
       ContentSettingsTypeSet content_type_set) override;
 
+  // content_settings::CookieSettings::Observer overrides:
+  void OnThirdPartyCookieBlockingChanged(
```

Comment:
I don't think having an observer for the UI is sufficient. If `kCookieControlsMode` is flipped by any mechanism outside of shields we need to make sure they stay in sync and don't end up with a case like kCookieControlsMode == allow and ContentSettingsType::COOKIES. I don't care if we update shields to match `kCookieControlsMode` or flip `kCookieControlsMode` back to match shields, but I want to make sure there is no way for them to get into an inconsistent or confusing state.

Reply:
yes, we the right way to solve this is with some "local policy" approach controlled by the current Shields state, but I haven't found a ready way to do this in Chromium, so it looks like we need to implement something on our own.